src/main/java/uk/nhs/tis/trainee/notifications/service/ProgrammeMembershipService.java

```diff
@@ -145,14 +145,21 @@ public boolean isExcluded(ProgrammeMembership programmeMembership) {
     }
 
     boolean hasMedicalSubType = curricula.stream()
+        .filter(c -> c.curriculumSubType() != null)
         .map(c -> c.curriculumSubType().toUpperCase())
         .anyMatch(INCLUDE_CURRICULUM_SUBTYPES::contains);
 
     boolean hasExcludedSpecialty = curricula.stream()
+        .filter(c -> c.curriculumSpecialty() != null)
         .map(c -> c.curriculumSpecialty().toUpperCase())
         .anyMatch(EXCLUDE_CURRICULUM_SPECIALTIES::contains);
 
-    return !hasMedicalSubType || hasExcludedSpecialty;
+    boolean hasSomeNonNullSpecialties = curricula.stream()
+        .anyMatch(c -> c.curriculumSpecialty() != null);
```

Comment:
This is included, but should be irrelevant. There are no programme memberships containing only null specialties that finished post 2024:

SELECT * FROM 
(SELECT pm.uuid, count(cm.id) curricula
FROM ProgrammeMembership pm
JOIN CurriculumMembership cm on cm.programmeMembershipUuid = pm.uuid
GROUP BY pm.uuid) pmsWithCmCount

JOIN

(SELECT pm.uuid, count(cm.id) curricula
FROM ProgrammeMembership pm
JOIN CurriculumMembership cm on cm.programmeMembershipUuid = pm.uuid
JOIN Curriculum c ON c.id = cm.curriculumId
WHERE c.specialtyId IS NULL
GROUP BY pm.uuid) pmsWithNullSpecialtyCount
ON pmsWithCmCount.uuid = pmsWithNullSpecialtyCount.uuid

JOIN

ProgrammeMembership pmfull ON pmsWithCmCount.uuid = pmfull.uuid

WHERE pmsWithCmCount.curricula = pmsWithNullSpecialtyCount.curricula
AND pmfull.programmeEndDate > '2013-01-01'

Reply:
(see: https://build.tis.nhs.uk/metabase/question/notebook#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJTRUxFQ1QgKiBGUk9NIChTRUxFQ1QgcG0udXVpZCwgY291bnQoY20uaWQpIGN1cnJpY3VsYVxuRlJPTSBQcm9ncmFtbWVNZW1iZXJzaGlwIHBtXG5KT0lOIEN1cnJpY3VsdW1NZW1iZXJzaGlwIGNtIG9uIGNtLnByb2dyYW1tZU1lbWJlcnNoaXBVdWlkID0gcG0udXVpZFxuR1JPVVAgQlkgcG0udXVpZCkgcG1zV2l0aENtQ291bnRcbkpPSU5cbihTRUxFQ1QgcG0udXVpZCwgY291bnQoY20uaWQpIGN1cnJpY3VsYVxuRlJPTSBQcm9ncmFtbWVNZW1iZXJzaGlwIHBtXG5KT0lOIEN1cnJpY3VsdW1NZW1iZXJzaGlwIGNtIG9uIGNtLnByb2dyYW1tZU1lbWJlcnNoaXBVdWlkID0gcG0udXVpZFxuSk9JTiBDdXJyaWN1bHVtIGMgT04gYy5pZCA9IGNtLmN1cnJpY3VsdW1JZFxuV0hFUkUgYy5zcGVjaWFsdHlJZCBJUyBOVUxMXG5HUk9VUCBCWSBwbS51dWlkKSBwbXNXaXRoTnVsbFNwZWNpYWx0eUNvdW50XG5PTiBwbXNXaXRoQ21Db3VudC51dWlkID0gcG1zV2l0aE51bGxTcGVjaWFsdHlDb3VudC51dWlkXG5KT0lOXG5Qcm9ncmFtbWVNZW1iZXJzaGlwIHBtZnVsbCBPTiBwbXNXaXRoQ21Db3VudC51dWlkID0gcG1mdWxsLnV1aWRcbldIRVJFIHBtc1dpdGhDbUNvdW50LmN1cnJpY3VsYSA9IHBtc1dpdGhOdWxsU3BlY2lhbHR5Q291bnQuY3VycmljdWxhXG5BTkQgcG1mdWxsLnByb2dyYW1tZUVuZERhdGUgPiAnMjAxMy0wMS0wMSciLCJ0ZW1wbGF0ZS10YWdzIjp7fX0sImRhdGFiYXNlIjozNn0sImRpc3BsYXkiOiJ0YWJsZSIsInZpc3VhbGl6YXRpb25fc2V0dGluZ3MiOnt9fQ==